Index: components/com_flexicontact/models/email.php
===================================================================
--- components/com_flexicontact/models/email.php	(revision 1162)
+++ components/com_flexicontact/models/email.php	(working copy)
@@ -137,15 +137,25 @@
 
 // validate the subject
 
-	if (($config_data->show_subject) and (empty($this->data->subject)))
+	if ($config_data->show_subject)
 		{
-		$ret = false;
-		$errors['subject'] = '<span class="fc_error">'.JText::_('COM_FLEXICONTACT_REQUIRED').'</span>';
+			$sub_error = false;
+			
+			if (($config_data->subject_as_list) and (!isset($config_data->subject_list_array[$this->data->subject])))
+				$sub_error = true;
+			elseif ((!$config_data->subject_as_list) and (empty($this->data->subject)))
+				$sub_error = true;
+			
+			if ($sub_error)
+				{
+				$ret = false;
+				$errors['subject'] = '<span class="fc_error">'.JText::_('COM_FLEXICONTACT_REQUIRED').'</span>';
+				}
 		}
 
 // validate the list selection
-
-	if (($config_data->list_opt == "mandatory") and (empty($this->data->list1)))
+	
+	if (($config_data->list_opt == "mandatory") and (!isset($config_data->list_array[$this->data->list1])))
 		{
 		$ret = false;
 		$errors['list'] = '<span class="fc_error">'.JText::_('COM_FLEXICONTACT_REQUIRED').'</span>';
@@ -348,12 +358,20 @@
 	else
 		$this->data->list_choice = '';
 
+// handle subject
+
+	$this->data->subject_value = $this->data->subject;
+	if ($config_data->show_subject && $this->data->subject != '' && $config_data->subject_as_list)
+		$this->data->subject_value = $config_data->subject_list_array[$this->data->subject];
+	
 // build the message to be sent to the site admin
 
 	$body = $this->email_merge($config_data->admin_template, $config_data);
 	jimport('joomla.mail.helper');
 	$clean_body = JMailHelper::cleanBody($body);
 	$clean_subject = JMailHelper::cleanSubject($this->data->subject);
+	if ($config_data->subject_as_list && $config_data->show_subject)
+		$clean_subject = $config_data->subject_list_array[$clean_subject];
 
 // build the Joomla mail object
 
Index: components/com_flexicontact/views/contact/view.html.php
===================================================================
--- components/com_flexicontact/views/contact/view.html.php	(revision 1163)
+++ components/com_flexicontact/views/contact/view.html.php	(working copy)
@@ -24,7 +24,6 @@
 		}
 
 // display the input form
-
 	echo "\n".'<div class="flexicontact">';
 	if (!empty($this->config_data->page_hdr))
 		echo "\n".'<h2>'.$this->config_data->page_hdr.'</h2>';
@@ -72,10 +71,17 @@
 // subject
 
 	if ($this->config_data->show_subject)
-		echo '<tr><td class="fc_prompt">'.JText::_('COM_FLEXICONTACT_SUBJECT').'</td>
-			<td class="fc_field">
-			<input type="text" name="subject" size="30" value="'.$this->escape($this->post_data->subject).'" /> '.
-				$this->errors['subject'].'</td></tr>';
+	{
+		echo '<tr><td class="fc_prompt">'.JText::_('COM_FLEXICONTACT_SUBJECT').'</td>';
+		echo '<td class="fc_field">';
+		
+		if ($this->config_data->subject_as_list)
+			echo Flexicontact_Utility::make_list('subject', $this->post_data->subject, $this->config_data->subject_list_array);
+		else
+			echo '<input type="text" name="subject" size="30" value="'.$this->escape($this->post_data->subject).'" />';
+		
+		echo ' ' . $this->errors['subject'] . '</td></tr>';
+	}
 
 // the select list
 
Index: administrator/components/com_flexicontact/models/log.php
===================================================================
--- administrator/components/com_flexicontact/models/log.php	(revision 1162)
+++ administrator/components/com_flexicontact/models/log.php	(working copy)
@@ -74,7 +74,7 @@
 			( NOW(), '.
 			$this->_db->Quote($email_data->from_name).','.
 			$this->_db->Quote($email_data->from_email).','.
-			$this->_db->Quote($email_data->subject).','.
+			$this->_db->Quote($email_data->subject_value).','.
 			$this->_db->Quote($email_data->area_data).','.
 			$this->_db->Quote($email_data->status_main).','.
 			$this->_db->Quote($email_data->status_copy).','.
Index: administrator/components/com_flexicontact/models/config.php
===================================================================
--- administrator/components/com_flexicontact/models/config.php	(revision 1163)
+++ administrator/components/com_flexicontact/models/config.php	(working copy)
@@ -38,7 +38,7 @@
 	$this->data = $component_params->toObject();
 
 // set defaults for all our parameters so that we have this all in one place
-
+	
 	if (!isset($this->data->logging))          $this->data->logging          = 0;
 	if (!isset($this->data->email_html))       $this->data->email_html       = 1;
 	if (!isset($this->data->autofill))         $this->data->autofill         = 'off';
@@ -49,6 +49,8 @@
 	if (!isset($this->data->show_copy))        $this->data->show_copy        = 1;
 	if (!isset($this->data->show_subject))     $this->data->show_subject     = 1;
 	if (!isset($this->data->default_subject))  $this->data->default_subject  = '';
+	if (!isset($this->data->subject_as_list))  $this->data->subject_as_list  = 0;
+	if (!isset($this->data->subject_list))     $this->data->subject_list     = '';
 	if (!isset($this->data->area_prompt))      $this->data->area_prompt      = '';
 	if (!isset($this->data->area_opt))         $this->data->area_opt         = 'optional';
 	if (!isset($this->data->area_width))       $this->data->area_width       = 40;
@@ -106,7 +108,22 @@
 		foreach ($this->data->list_array as &$list_item)
 			$this->translate($list_item);
 		}
+
+// subject as list
+	if ($site)
+	{
+		$subject_list = $this->data->subject_list;
+		$subject_list = strtr($subject_list,"\r","");
+		$subject_list = strtr($subject_list,"\n","");
 		
+		$this->data->subject_list_array = explode(",",$subject_list);
+		$this->data->subject_list_count = count($this->data->subject_list_count);
+		
+		foreach ($this->data->subject_list_array as &$list_item) {
+			$this->translate($list_item);
+		}
+	}
+	
 	return $this->data;
 }
 
@@ -115,6 +132,8 @@
 //
 function translate(&$value)
 {
+	$value = trim($value);
+	
 	if ( ($value != '') and ($value{0} == '_') )
 		$value = JText::_('COM_FLEXICONTACT'.$value);
 }
@@ -145,6 +164,8 @@
 		case 'config_fields':
 			$post_data['show_subject'] = JRequest::getVar('show_subject');			// radio button
 			$post_data['default_subject'] = JRequest::getVar('default_subject');
+			$post_data['subject_as_list'] = JRequest::getVar('subject_as_list');	// radio button
+			$post_data['subject_list'] = JRequest::getVar('subject_list');
 			$post_data['show_copy'] = JRequest::getVar('show_copy');				// radio button
 			$post_data['area_prompt'] = JRequest::getVar('area_prompt');
 			$post_data['area_opt'] = JRequest::getVar('area_opt');
@@ -216,7 +237,7 @@
 		$id = $table->find(array('element' => LAFC_COMPONENT));
 		$table->load($id);
 		}
-		
+	
 	$table->bind($data);
 	return $table->store();
 
Index: administrator/components/com_flexicontact/views/config_fields/view.html.php
===================================================================
--- administrator/components/com_flexicontact/views/config_fields/view.html.php	(revision 1162)
+++ administrator/components/com_flexicontact/views/config_fields/view.html.php	(working copy)
@@ -47,7 +47,20 @@
 		echo '<td><input type="text" size="40" name="default_subject" value="'.$this->config_data->default_subject.'" /> '.
 			Flexicontact_Utility::make_info(JText::_('COM_FLEXICONTACT_V_DEFAULT_SUBJECT_DESC')).'</td>';
 	echo "\n</tr>";
-	
+
+// use subject as a list
+
+	echo "\n<tr>";
+		echo '<td align="right">'.JText::_('COM_FLEXICONTACT_V_USE_SUBJECT_AS_LIST').'</td>';
+		echo '<td>'.Flexicontact_Utility::make_radio('subject_as_list',$this->config_data->subject_as_list).'</td>';
+		echo '<td></td>';
+		echo '<td valign="top" align="right">'.JText::_('COM_FLEXICONTACT_V_LIST_ITEMS').'</td>';
+		echo '<td valign="top"><textarea rows="3" cols="40" name="subject_list">'.$this->config_data->subject_list.'
+			</textarea>'.'</td>';
+		echo '<td></td>';
+		echo '<td></td>';
+	echo "\n</tr>";
+
 // send a copy to me checkbox	
 
 	echo "\n<tr>";
Index: administrator/language/en-GB/en-GB.com_flexicontact.ini
===================================================================
--- administrator/language/en-GB/en-GB.com_flexicontact.ini	(revision 1162)
+++ administrator/language/en-GB/en-GB.com_flexicontact.ini	(working copy)
@@ -6,6 +6,10 @@
 ; Licence	: GNU General Public License
 ; Please ensure all language files are saved as UTF8 NO BOM
 ;
+
+; new
+COM_FLEXICONTACT_V_USE_SUBJECT_AS_LIST="Use subject as list"
+
 ; new in v5.00
 
 COM_FLEXICONTACT_CONFIG_NAME="Configuration Name"
